pkg/syncer/metadatasyncer.go

```diff
@@ -397,21 +415,46 @@ func pvUpdated(oldObj, newObj interface{}, metadataSyncer *metadataSyncInformer)
 	}
 	log.Debugf("PVUpdated: PV Updated from %+v to %+v", oldPv, newPv)
 
-	// Verify if pv is a vsphere csi volume
-	if oldPv.Spec.CSI == nil || newPv.Spec.CSI == nil || newPv.Spec.CSI.Driver != csitypes.Name {
-		log.Debugf("PVUpdated: PV is not a Vsphere CSI Volume: %+v", newPv)
-		return
-	}
 	// Return if new PV status is Pending or Failed
 	if newPv.Status.Phase == v1.VolumePending || newPv.Status.Phase == v1.VolumeFailed {
 		log.Debugf("PVUpdated: PV %s metadata is not updated since updated PV is in phase %s", newPv.Name, newPv.Status.Phase)
 		return
 	}
-	// Return if labels are unchanged
-	if (oldPv.Status.Phase == v1.VolumeAvailable || oldPv.Status.Phase == v1.VolumeBound) && reflect.DeepEqual(newPv.GetLabels(), oldPv.GetLabels()) {
-		log.Debugf("PVUpdated: PV labels have not changed")
-		return
+	// Verify if csi migration is ON and check if there is any label update or migrated-to annotation was received for the PV
+	if metadataSyncer.configInfo.Cfg.FeatureStates.CSIMigration && newPv.Spec.VsphereVolume != nil {
+		if oldPv.Status.Phase == v1.VolumeBound &&
+			reflect.DeepEqual(newPv.GetAnnotations(), oldPv.GetAnnotations()) &&
+			reflect.DeepEqual(newPv.Labels, oldPv.Labels) {
+			log.Debugf("PVUpdated: PV labels and annotations have not changed")
+			return
+		}
+		// Check if there is a new annotation on the PV
+		if !reflect.DeepEqual(newPv.GetAnnotations(), oldPv.GetAnnotations()) {
+			provisionerAnnotation, ok := oldPv.GetAnnotations()["provisioned-by"]
+			if ok {
+				if provisionerAnnotation == inTreePluginName {
+					_, ok := oldPv.GetAnnotations()["pv.kubernetes.io/migrated-to"]
```

Comment:
When we observe  `pv.kubernetes.io/migrated-to: csi.vsphere.vmware.com` annotation is added, we need to register volume with CNS? I could not find code here in this func `pvUpdated which calls VMS func to register volume with CNS.

For statically created volume, syncer needs to register disk with CNS.

Reply:
It is taken care of, by this https://github.com/kubernetes-sigs/vsphere-csi-driver/pull/264/files#diff-6fc246216efb3eb9047f82d1116e9813R657, right ?